test/widgets/trip_page_builder_test.dart

```diff
@@ -0,0 +1,270 @@
+import '../util/pump_material.dart';
+import '../util/request_processor.dart';
+import '../util/request_processor.mocks.dart';
+
+void main() {
+  late List<Map<String, dynamic>> drives;
+  late Profile profile;
+  final MockRequestProcessor processor = MockRequestProcessor();
+
+  setUpAll(() async {
+    MockServer.setProcessor(processor);
+  });
+  setUp(() async {
+    reset(processor);
+    profile = ProfileFactory().generateFake(id: 1);
+    SupabaseManager.setCurrentProfile(profile);
+  });
+
+  // since the code for the TripPageBuilder is the same for both pages,
+  // we only test the DrivesPage and assume that the RidesPage works the same.
+  // This test is to show that if the ridesPage is called there are rideCards shown.
+  testWidgets('shows Ride card for RidesPage', (WidgetTester tester) async {
+    final List<Map<String, dynamic>> rides = [
+      RideFactory()
+          .generateFake(
+            id: 1,
+            endTime: DateTime.now().add(const Duration(days: 1)),
+            hideInListView: false,
+            rider: NullableParameter(null),
+            drive: NullableParameter(null),
+            driveId: 3,
+            createDependencies: false,
+          )
+          .toJsonForApi(),
+      RideFactory()
+          .generateFake(
+            id: 2,
+            endTime: DateTime.now().add(const Duration(days: 1)),
+            hideInListView: false,
+            rider: NullableParameter(null),
+            drive: NullableParameter(null),
+            driveId: 4,
+            createDependencies: false,
+          )
+          .toJsonForApi(),
+    ];
+    whenRequest(processor, urlMatcher: equals('/rest/v1/rides?select=%2A&rider_id=eq.1&order=start_time.asc.nullslast'))
+        .thenReturnJson(rides);
+    whenRequest(processor, urlMatcher: matches(RegExp(r'/rest/v1/drives.*id=eq\.3')), methodMatcher: equals('GET'))
+        .thenReturnJson(DriveFactory().generateFake(id: 1).toJsonForApi());
+    whenRequest(processor, urlMatcher: matches(RegExp(r'/rest/v1/drives.*id=eq\.4')), methodMatcher: equals('GET'))
+        .thenReturnJson(DriveFactory().generateFake(id: 2).toJsonForApi());
+
+    await pumpMaterial(tester, const RidesPage());
+    await tester.pump();
+    final Finder rideCard = find.byType(RideCard);
+    expect(rideCard, findsNWidgets(2));
+  });
+
+  group('tabs', () {
+    setUp(() async {});
+    testWidgets('finds both Tabs(before and after Drives are loaded)', (WidgetTester tester) async {
+      whenRequest(processor,
+              urlMatcher: equals('/rest/v1/drives?select=%2A&driver_id=eq.1&order=start_time.asc.nullslast'))
+          .thenReturnJson([]);
+      await pumpMaterial(tester, const DrivesPage());
+      final Finder tabBar = find.byType(TabBar);
+      expect(tabBar, findsOneWidget);
+      expect(find.descendant(of: tabBar, matching: find.byType(Tab)), findsNWidgets(2));
+      await tester.pump();
+      final Finder tabBarLoaded = find.byType(TabBar);
```

Comment:
you don't need to redefine the finder, just write another expectation

Reply:
yes okay wasn't sure